qiskit/test/mock/fake_backend.py

```diff
@@ -38,6 +62,257 @@ def __init__(self, token="123456", url="https://"):
+            )
+
+    @property
+    def max_circuits(self):
+        return None
+
+    @classmethod
+    def _default_options(cls):
+        if _optionals.HAS_AER:
+            from qiskit.providers import aer
+
+            return aer.QasmSimulator._default_options()
+        else:
+            return basicaer.QasmSimulatorPy._default_options()
+
+    @property
+    def dtm(self) -> float:
+        """Return the system time resolution of output signals
+        Returns:
+            dtm: The output signal timestep in seconds.
+        """
+        return self._conf_dict.get("dtm")
+
+    @property
+    def meas_map(self) -> List[List[int]]:
+        """Return the grouping of measurements which are multiplexed
+        This is required to be implemented if the backend supports Pulse
+        scheduling.
+        Returns:
+            meas_map: The grouping of measurements which are multiplexed
+        """
+        return self._conf_dict.get("meas_map")
+
+    def qubit_properties(
+        self, qubit: Union[int, List[int]]
+    ) -> Union[QubitProperties, List[QubitProperties]]:
+        """Return QubitProperties for a given qubit.
+        Args:
+            qubit: The qubit to get the
+                :class:`~qiskit.provider.QubitProperties` object for. This can
+                be a single integer for 1 qubit or a list of qubits and a list
+                of :class:`~qiskit.provider.QubitProperties` objects will be
+                returned in the same order
+        """
+        if not self._qubit_properties:
+            self._qubit_properties = qubit_props_dict_from_props_dict(self._props_dict)
+        if isinstance(qubit, int):  # type: ignore[unreachable]
+            return self._qubit_properties.get(qubit)
+        if isinstance(qubit, List):
+            return [self._qubit_properties.get(q) for q in qubit]
+        return None
+
+    def run(self, run_input, **kwargs):
+        """Main job in simulator"""
+        circuits = run_input
+        pulse_job = None
+        if isinstance(circuits, (pulse.Schedule, pulse.ScheduleBlock)):
+            pulse_job = True
+        elif isinstance(circuits, circuit.QuantumCircuit):
+            pulse_job = False
+        elif isinstance(circuits, list):
+            if circuits:
+                if all(isinstance(x, (pulse.Schedule, pulse.ScheduleBlock)) for x in circuits):
+                    pulse_job = True
+                elif all(isinstance(x, circuit.QuantumCircuit) for x in circuits):
+                    pulse_job = False
+        if pulse_job is None:
+            raise QiskitError(
+                "Invalid input object %s, must be either a "
+                "QuantumCircuit, Schedule, or a list of either" % circuits
+            )
+        if _optionals.HAS_AER:
+            from qiskit.providers import aer
+
+            if pulse_job:
+                from qiskit.providers.aer.pulse import PulseSystemModel
+
+                system_model = PulseSystemModel.from_backend(self)
```

Comment:
Does this work? I assumed it would need to be able to pull the instruction schedule map via `self.defaults` which doesn't exist here

Reply:
I don't think this part works, but I haven't tested it. I was thinking to fix this in a separate PR in Aer https://github.com/Qiskit/qiskit-aer/issues/1478 because I don't want to add too much Aer code to Terra and the PR is already getting very complicated. But if you think it's a good idea to support pulse now I can do that. If not maybe we can raise a warning / error here that pulse simulation is not supported yet. What do you think?

Reply:
I think just raising an error on a pulse schedule input to `run()` for now is fine. After we add support for backendv2 to the pulse simulator in Aer we can add support for it here.

Reply:
Pulse job now raise a `QiskitError`